tests/unit/lms/views/admin/lti_registration_test.py

```diff
@@ -0,0 +1,156 @@
+from unittest.mock import sentinel
+
+import pytest
+from sqlalchemy.exc import IntegrityError
+
+from lms.views.admin.lti_registration import AdminLTIRegistrationViews
+from tests import factories
+from tests.matchers import Any, temporary_redirect_to
+
+
+@pytest.mark.usefixtures(
+    "pyramid_config", "application_instance_service", "lti_registration_service"
+)
+class TestAdminApplicationInstanceViews:
+    def test_registrations(self, views):
+        assert views.registrations() == {}
+
+    def test_registration_new(self, views):
+        assert views.registration_new() == {}
+
+    def test_registration_new_post(
+        self, pyramid_request_new_registration, lti_registration_service
+    ):
+        response = AdminLTIRegistrationViews(
+            pyramid_request_new_registration
+        ).registration_new_post()
```

Comment:
```suggestion
    @pytest.mark.usefixture('with_form_submission')
    def test_registration_new_post(self, views, lti_registration_service):
        response = views.registration_new_post()
```

If we use `pyramid_request_new_registration` to modify the request instead of relying on it to return it, we can use the same fixture in each test. This will cut down on some repetition and make the tests easier to read.

I think this is true for all the tests here which modify the request before constucting their own view instance. I think you can construct the standard one, and then modify the request after. It will still reference it.